### PR TITLE
[#5162] Add ability to add roles to users and groups in the Gravitino CLI

### PR DESCRIPTION
**Pull request by justinmclean**
### What changes were proposed in this pull request?

CLI commands and code to add roles to users and groups.

### Why are the changes needed?

To support adding and removing roles.

Fix: #5162

### Does this PR introduce _any_ user-facing change?

No, but it adds to the CLI.

### How was this patch tested?

Compiled and tested locally.

### CODE REVIEW

**Comment by justinmclean** [comment]
@shaofengshi this is walting on teh role PR to be merged or you can review the add user or groups to roles part

**Comment by justinmclean** [comment]
@shaofengshi  this is now ready to review

File: docs/cli.md [hunk 1]
```diff
@@ -523,10 +523,33 @@ gcli role list
...
 
 ```bash
 gcli role delete --role admin
- ```
+```
+
+#### Add a role to a user
+
+```bash
+gcli user update --metalake metalake_demo --user new_user --role admin
```
Comment (shaofengshi) [inline docs/cli.md:537]: Can we use "grant" / "revoke" as the command to operate the role to user&groups? "update"/"delete" is not that accurate for this kind of operations.

File: docs/cli.md [hunk 2]
```diff
@@ -523,10 +523,33 @@ gcli role list
...
+```
+
+#### Remove a role from a user
+
+```bash
+gcli user delete --metalake metalake_demo --user new_user --role admin
```
Comment (shaofengshi) [inline docs/cli.md:543]: Btw, please remove the "--metalake" from the samples.

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/AddRoleToUser.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  public AddRoleToUser(
+      String url, boolean ignoreVersions, String metalake, String user, String role) {
+    super(url, ignoreVersions);
+    this.metalake = metalake;
+    this.user = user;
+    this.role = role;
+  }
+
+  /** Adds a role to a user. */
+  public void handle() {
```
Comment (shaofengshi) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/AddRoleToUser.java:54]: also here.

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/SetOwner.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.MetadataObjects;
+import org.apache.gravitino.authorization.Owner;
+import org.apache.gravitino.cli.CommandEntities;
+import org.apache.gravitino.cli.ErrorMessages;
+import org.apache.gravitino.client.GravitinoClient;
+import org.apache.gravitino.exceptions.NoSuchMetadataObjectException;
+import org.apache.gravitino.exceptions.NoSuchMetalakeException;
+
+public class SetOwner extends Command {
```
Comment (shaofengshi) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/SetOwner.java]: This file seems not related with this issue and PR.

File: docs/cli.md (hunk 1 above)
Comment (justinmclean) [reply docs/cli.md:537]: Sure I went this was as Jerry prefered a "less is more approach"

File: docs/cli.md (hunk 2 above)
Comment (justinmclean) [reply docs/cli.md:543]: Done

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/SetOwner.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      this.entityType = MetadataObject.Type.TABLE;
+    } else if (entityType.equals(CommandEntities.COLUMN)) {
+      this.entityType = MetadataObject.Type.COLUMN;
+    } else {
+      this.entityType = null;
+    }
+  }
+
+  /** Sets the owner of an entity. */
+  public void handle() {
```
[the remark this answers is not in the document]
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/SetOwner.java]: Done

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/SetOwner.java (hunk 4 above)
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/SetOwner.java]: That file has already been removed before you reviewed the PR - please take note when it says "outdated"

**Comment by justinmclean** [comment]
@shaofengshi ready again for you to review

**Review by shaofengshi — APPROVED** [review]
LGTM
